raster-foundry/raster-foundry
# Remove Scapegoat errors for API project

## Overview

Remedy the warning messages and use `SuppressWarnings` where a straightforward solution isn't obvious. Also, clean up imports across all files touched.

### Checklist

~- [ ] Styleguide updated, if necessary~
~- [ ] [Swagger specification](https://github.com/raster-foundry/raster-foundry-api-spec) updated~
~- [ ] Symlinks from new migrations present or corrected for any new migrations~


~- [ ] Any content changes are properly templated using `BUILDCONFIG.APP_NAME`~
~- [ ] Any new SQL strings have tests~

### Demo

<img width="1440" alt="screen shot 2018-08-21 at 00 18 27" src="https://user-images.githubusercontent.com/43639/44380161-be27bf80-a4d7-11e8-99e5-ce6c390a1147.png">

### Notes

There is still one **Match instead of partial function** that I was unable to make go away in `scene.Routes.scala:335`.


## Testing Instructions

Execute the following command and confirm the Scapegoat output:

```bash
$ docker-compose run --rm --no-deps api-server api/scapegoat

...

[info] [scapegoat] Removing scapegoat class directory: /opt/raster-foundry/app-backend/api/target/scala-2.11/scapegoat-classes
[info] [scapegoat] setting output dir to [/opt/raster-foundry/app-backend/api/target/scala-2.11/scapegoat-report]
[info] [scapegoat] ignored file patterns: .*/datamodel/.*
[info] Compiling 74 Scala sources to /opt/raster-foundry/app-backend/api/target/scala-2.11/scapegoat-classes ...
[info] [info] [scapegoat] 115 activated inspections
[info] [info] [scapegoat] List(.*/datamodel/.*) ignored file patterns
[info] scene.Routes.scala:335: Match instead of partial function
          A map match can be replaced with a partial function for greater readability: com.azavea.rf.common.utils.CogUtils.thumbnail(uri, thumbnailParams.width, thumbnailParams.height, thumbnailParams.red, thumbnailParams.green, thumbnailParams.blue, thumbnailParams.floor)(scala.concurrent.ExecutionContext.Implicits.global).map[akka.http.scaladsl.model.HttpEntity.Strict](((tile: geotrellis.raster.MultibandTile) => tile match {
  case (t @ _) if t.bands.length.>=(3) => akka.http.scaladsl.model.HttpEntity.apply(model.this.ContentType.apply(akka.http.scaladsl.model.MediaTypes.image/p

[info] /opt/raster-foundry/app-backend/api/src/main/scala/scene/Routes.scala:335:24: Match instead of partial function
[info]                   ).map(
[info]                        ^
[info] [info] [scapegoat] Analysis complete: 74 files - 0 errors 0 warns 1 infos
[info] [info] [scapegoat] Written HTML report [/opt/raster-foundry/app-backend/api/target/scala-2.11/scapegoat-report/scapegoat.html]
[info] [info] [scapegoat] Written XML report [/opt/raster-foundry/app-backend/api/target/scala-2.11/scapegoat-report/scapegoat.xml]
[info] [info] [scapegoat] Written Scalastyle XML report [/opt/raster-foundry/app-backend/api/target/scala-2.11/scapegoat-report/scapegoat-scalastyle.xml]

...
```

Connects https://github.com/raster-foundry/raster-foundry/issues/3758
